src/Application.js

```diff
@@ -180,6 +180,10 @@ export class Application {
  */
 Application._plugins = [];
 
+/**
+ * @typedef {typeof import('./plugins/ApplicationPlugin.js').ApplicationPlugin} ApplicationPlugin
```

Comment:
These typedef lines make me think that the .d.ts files are built from the JSDocs - should that process be added to package.json?

Reply:
Unfortunately that is not the case, this is just so I can use proper jsdoc types in other files without importing the file